src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/acls/gcs/GcsAuthorizationProvider.scala

```diff
@@ -0,0 +1,95 @@
+
+package org.broadinstitute.dsde.agora.server.dataaccess.acls.gcs
+
+import akka.actor._
+import scala.concurrent.Future
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.broadinstitute.dsde.agora.server.{AgoraConfig, GoogleCredentialHandler}
+import org.broadinstitute.dsde.agora.server.dataaccess.acls._
+import org.broadinstitute.dsde.agora.server.business.AgoraBusiness
+import AgoraPermissions._
+import org.broadinstitute.dsde.agora.server.model.{GoogleAccessControls, AgoraEntity}
+import GcsApiJsonSupport._
+import spray.client.pipelining._
+import spray.http.OAuth2BearerToken
+import spray.http.StatusCodes._
+import spray.http.HttpResponse
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+object GcsAuthorizationProvider extends AuthorizationProvider with LazyLogging {
+  implicit val actorSystem = ActorSystem("agora")
+  import actorSystem.dispatcher
+
+  // GCS sends back a malformed ETag header according to HTTP standards. Spray complains. We currently don't care about ETag, so filter it out.
+  val removeEtagHeaders: HttpResponse => HttpResponse =
+    r => r.withHeaders(r.headers.filter(!_.name.startsWith("ETag")))
+
+  val pipeline = addCredentials(OAuth2BearerToken(GoogleCredentialHandler.accessToken)) ~> sendReceive ~> removeEtagHeaders
+
+  private case class GoogleResponse(entity: AgoraEntity, response: HttpResponse)
+
+  def namespaceToBucketName(namespace: String): String = "agoraNamespace_" + namespace
+
+  def entityToObjectName(entity: AgoraEntity): String = "agoraEntity_" + entity.entityType.get + "_" + entity.name.get + "_" + entity.snapshotId.get
+
+  def getGcsObjectRoleRequestUrl(agoraEntity: AgoraEntity, userEmail: String): String = {
+    val agoraBusiness = new AgoraBusiness
+    agoraBusiness.hasNamespaceNameId(agoraEntity) match {
+      case true =>
+        "https://www.googleapis.com/storage/v1/b/" + namespaceToBucketName(agoraEntity.namespace.get) + "/o/" + entityToObjectName(agoraEntity) + "/acl/user-" + userEmail
```

Comment:
lets yank out the google api url and put it in a constant. When the rev the api it will make changing it easier.